core/insertion_marker_manager.js

```diff
@@ -157,14 +158,21 @@ Blockly.InsertionMarkerManager.prototype.dispose = function() {
   this.availableConnections_.length = 0;
   this.closestConnection_ = null;
   this.localConnection_ = null;
-  if (this.firstMarker_) {
-    this.firstMarker_.dispose();
-    this.firstMarker_ = null;
-  }
-  if (this.lastMarker_) {
-    this.lastMarker_.dispose();
-    this.lastMarker_ = null;
+
+  Blockly.Events.disable();
+  try {
+    if (this.firstMarker_) {
+      this.firstMarker_.dispose();
+      this.firstMarker_ = null;
+    }
+    if (this.lastMarker_) {
+      this.lastMarker_.dispose();
+      this.lastMarker_ = null;
+    }
+  } finally {
```

Comment:
Yup!  You should use try/finally almost any time that you're disabling events in blockly, because event disable/enable isn't a boolean.  It's a counter, so you need to have symmetrical disables and enables.  If an exception is throw between the disable and the enable and there's no `finally` clause, you may never get events again.